gravitational/webapps
# Allow users to update Upgrade Window Start

This PR adds a prop `children` to the "Help & Support" page that allows enterprise to extend it. `webapps.e` will have its own page that will add a new section to Support with some cloud specific information. Some components of `Support` have been made more generic and exported so we can use them on enterprise.

This PR also updates the events table to properly show the new audit event type.

Enterprise PR using this feature: https://github.com/gravitational/webapps.e/pull/357

## Testing

To run the whole thing, you can:

1. Run salescenter locally (follow [this](https://github.com/gravitational/cloud#sales-center))
2. Checkout `master` ~`mcbattirola/add-cloud-upgrade-windows`~ on both `teleport` and `teleport.e`, and `webapps.e` on branch `mcbattirola/add-cloud-upgrade-windows`
3. comment [this line](https://github.com/gravitational/teleport/blob/master/lib/modules/modules.go#L130) on `lib/modules/modules.go` so we can run cloud without second factor
4. build and run **teleport.e** with [this](https://github.com/gravitational/cloud/blob/master/db/fixtures/license.pem) license (under auth_service in the config file). I recommend you copy `cloud/dev/teleport/config.yaml` and add:

```yaml
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  tokens:
  - proxy,node:[token]
  authentication:
  license_file: /home/matheus/tc/db/fixtures/license.pem
```

You have to run it like this:

```bash
TELEPORT_CLOUD_HOSTPORT="api.localhost:3080" DEBUG=1 /path/to/teleport/e/build/teleport start -c my-config-file.yaml -d --insecure
```

6. Run this branch of the teleport UI with `yarn start-teleport-e --target=https://localhost:4080/web`
Then you should be able to make a request from the local teleport to the salescenter

## Screenshots

![Screenshot from 2022-07-14 12-23-57](https://user-images.githubusercontent.com/8400114/179018581-4a36bd8e-9a7d-4b9d-88a0-db90daf9ed23.png)


![Screenshot from 2022-07-19 14-00-08](https://user-images.githubusercontent.com/8400114/179807704-aa25cc24-9490-455e-8e8d-88f05be06f3a.png)

Closes https://github.com/gravitational/cloud/issues/1808
